Messages translation keys

darrenvong opened the pull request:
<!---
Please describe the pull request below.
If it closes an issue, make sure to write "closes Couchers-org/couchers#1234"
If there is an issue but it isn't completely closed, still refer to the issue number, eg. "part of Couchers-org/couchers#1234"
--->
This is probably the most heavily hard-coded and thus painful one yet... not to mention some pretty hairy/gross time interval logic I've yet to deal with...

<!---
Checklists - you can remove one that is not applicable (ie. remove backend checklist if you only worked on the web frontend)
If you need help with any of these, please ask :)
--->

**Web frontend checklist**
- [x] Formatted my code with `yarn format && yarn lint --fix`
- [x] There are no warnings from `yarn lint`
- [ ] There are no console warnings when running the app
- [ ] Added any new components to storybook
- [x] Added tests where relevant
- [x] All tests pass
- [x] Clicked around my changes running locally and it works
- [ ] Checked Desktop, Mobile and Tablet screen sizes


<!---
Remember to request review from couchers-org/web, couchers-org/@backend or an individual.
Once your code is approved, remember to merge it if you have write access
--->

darrenvong committed:
refactor: i18n admins dialog and CreateGroupChat dialog

darrenvong committed:
refactor: i18n rest of components in groupchats folder

darrenvong committed:
refactor: i18n control message

This is an indicator of worse to come with the time
interval component... it's gonna be a right bitch to do...

darrenvong committed:
refactor: i18n message list component

darrenvong commented on app/web/features/messages/messagelist/ControlMessageView.tsx:
```diff
@@ -41,8 +42,7 @@ export default function ControlMessageView({
   );
   const { ref } = useOnVisibleEffect(onVisible);
 
-  const isCurrentUser = author?.userId === currentUserId;
-  const authorName = isCurrentUser ? "you" : firstName(author?.name);
```
I decided to get rid of this "you" thing to simplify the i18n logic so even though I created the chat, it will say "Darren created the chat" rather than "You created the chat". Otherwise I'd have to duplicate all types of control messages just for the current user as a separate variant, since we can't reliably assume we can just swap out name for "you" as in English. Not worth the pain and complexity to keep this tiny thing imo.

lucaslcode commented on app/web/features/messages/locales/en.json:
```diff
@@ -1,35 +1,108 @@
+  "title": "Title",
+  "write_reference": "Write a Reference",
+  "close_request_dialog_title": "Are you done messaging?",
+  "close_request_dialog_host": "Please make sure you are done chatting before you    reject their request.",
+  "close_request_dialog_surfer": "Please make sure you are done chatting before you    cancel your request.",
```
```suggestion
  "close_request_dialog_host": "Please make sure you are done chatting before you reject their request.",
  "close_request_dialog_surfer": "Please make sure you are done chatting before you cancel your request.",
```

lucaslcode commented on app/web/features/messages/utils.ts:
```diff
@@ -17,6 +18,53 @@ export function messageTargetId(message: Message.AsObject) {
+}
+
+/**
+ * @deprecated Use controlMessage instead
```
Is it still used?

darrenvong replied:
It's in host requests for now, but might not need to do this if I finish converting the keys there in this PR later on as I can just delete the old function

darrenvong committed:
refactor: remove use of old controlMessageText fn

To avoid the need of deprecation

darrenvong committed:
refactor: use the new translationStaticProps for loading in translations

darrenvong committed:
refactor: i18n display of time intervals

Ideally this should be using the RelativeTime from the Intl API, but the
browser support for that is not great - the baseline is
Safari 14+ (2020 browser) which is too recent and so many users will
likely see errors if they are on iOS.

Because of that, I decide to stick with our own timeAgo function, but
i18n it with i18next's t function passed in instead.

darrenvong committed:
refactor: i18n host request components

darrenvong committed:
fix: bad merge that omitted "nav.mission" key

darrenvong committed:
refactor: i18n rest of host request components

darrenvong committed:
chore: add in languages placeholder file we currently support

So that we don't have to manually click through Weblate to get them
showing up

darrenvong commented on app/web/utils/timeAgo.ts:
```diff
@@ -19,6 +21,9 @@ export interface FuzzySpec {
 }
 
+/**
+ * @deprecated Use `timeAgoI18n` instead.
```
Had to do this since `profile` feature area still uses this function and I don't fancy dealing with this for now as part of this...

darrenvong commented:
@lucaslcode This was painful but I think finally it's ready for another look now 😵

darrenvong committed:
fix: use MUI's `titleAccess` prop instead to improve icon accessibility

lucaslcode reviewed (approved):
Amazing work

darrenvong commented:
Dropping Couchers-org/couchers#4152 here so it links back to the main issue
